Changelog for the release of WordPressCS 3.4.0 (#2758)
WordPress-Coding-Standards/WordPress-Coding-Standards: CHANGELOG.md

Reviewer: ```suggestion
This project adheres to [Semantic Versioning](https://semver.org/) and [Keep a CHANGELOG](https://keepachangelog.com/).
```

Author: OMG.... that typo has been in there since this file was created in 2014... Nice catch!

Reviewer: Yeah, I started reading and was like: hmmmm this is odd 😅

_No documentation available about unreleased changes as of yet._

## [3.4.0] - 2026-07-13

We're happy to welcome [@rodrigoprimo] as co-maintainer of WordPressCS as of this release.

### Added

> - `WordPress.Arrays.ArrayDeclarationSpacing`: new `allow_single_item_single_line_explicit_key_arrays` property to replace the, no deprecated, `allow_single_item_single_line_associative_arrays` property. [#2696]

Reviewer 2: ```suggestion
- `WordPress.Arrays.ArrayDeclarationSpacing`: new `allow_single_item_single_line_explicit_key_arrays` property to replace the, now deprecated, `allow_single_item_single_line_associative_arrays` property. [#2696]
```

- Improved the wording of the error message for `WordPress.PHP.RestrictedPHPFunctions`. [#2702]
- Various housekeeping, including documentation and test improvements. Includes a contribution by [@dd32].

### Deprecated
- `WordPress.Arrays.ArrayDeclarationSpacing`: the `allow_single_item_single_line_associative_arrays` property has been deprecated in favor of the new `allow_single_item_single_line_explicit_key_arrays` property. [#2696]
    This is a name change only. The functionality of these properties is the same.

### Fixed
- `WordPress.DB.PreparedSQL` and `WordPress.DB.PreparedSQLPlaceholders`: false positive for static method calls to a non-global class named `wpdb`. [#2753]
- `WordPress.Security.EscapeOutput`: false positive for `get_search_query()` when the `$escaped` parameter was passed as fully qualified or non-lowercase `true`. [#2618]
- `WordPress.WP.AlternativeFunctions`: false negative when class functions/constants/properties use the same name as select global WP constants/functions. [#2617]
- `WordPress.WP.AlternativeFunctions`: false positive for fully qualified references to the global PHP stream constants \STDIN, \STDOUT, and \STDERR. [#2617]

> - `WordPress.WP.CronInterval`: false positive when the callback function reference used a different case than the function declarationeven though they are in the same file. [#2730]

Reviewer 2: ```suggestion
- `WordPress.WP.CronInterval`: false positive when the callback function reference used a different case than the function declaration even though they are in the same file. [#2730]
```

### Deprecated
- `WordPress.Arrays.ArrayDeclarationSpacing`: the `allow_single_item_single_line_associative_arrays` property has been deprecated in favor of the new `allow_single_item_single_line_explicit_key_arrays` property. [#2696]
    This is a name change only. The functionality of these properties is the same.

### Fixed
- `WordPress.DB.PreparedSQL` and `WordPress.DB.PreparedSQLPlaceholders`: false positive for static method calls to a non-global class named `wpdb`. [#2753]
- `WordPress.Security.EscapeOutput`: false positive for `get_search_query()` when the `$escaped` parameter was passed as fully qualified or non-lowercase `true`. [#2618]
- `WordPress.Security.EscapeOutput`: false negative for `_deprecated_file()` calls when the `basename( __FILE__ )` pattern used non-standard casing for either `basename()` and/or `__FILE__`. [#2729]
- `WordPress.WP.AlternativeFunctions`: false negative when class functions/constants/properties use the same name as select global WP constants/functions. [#2617]
- `WordPress.WP.AlternativeFunctions`: false positive for fully qualified references to the global PHP stream constants \STDIN, \STDOUT, and \STDERR. [#2617]

> - `WordPress.WP.CronInterval`: false positive when the callback function reference used a different case than the function declaration even though they are in the same file. [#2730]

Reviewer: ```suggestion
- `WordPress.WP.CronInterval`: false positive when the callback function reference used a different case than the function declaration, even though they are in the same file. [#2730]
```
